### PR TITLE
Remove unused ESM exports from `@directus/shared`

### PR DESCRIPTION
**Pull request by rijkvanzanten**
The "ESM" build of `@directus/shared` wasn't actually valid ESM, and has subsequently never really been used in any of the ESM packages we have. The main package.json file defaults `.js` files to CJS (because of the omission of `"type": "module"`), which in turn means that Node requires the ESM files to be `.mjs`. Because those were missing, it'd end up in a weird in-between state. The "ESM" build isn't valid ESM, ESM tooling wasn't trying to use the ESM builds, and the App ended up using the ESM build, but still using TS / Rollups resolution magic (which isn't native ESM).

Long story short, nightmare! I tried resolving this by building the shared stuff to "proper ESM", but that was causing issues in packages that were relying on the wrong ESM export, so decided that it's best to have shared be just proper CJS for now, until everything else is in ESM

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
cc @paescuj

**Comment by rijkvanzanten** [comment]
Oh man this is even more confusing than I thought. `esbuild` (used by `vite`) relies on the fake ESM export to know what to externalize, but Vite itself by default uses CJS because the app project is CJS, so you need both available, but then native ESM things go bang as the exported ESM isn't... valid ESM ☠️

Back to the drawing board..

**Comment by rijkvanzanten** [comment]
Update: with the eye on everything above, I think it makes the most sense to make a new ESM-only `@directus/utils` package that will effectively replace `@directus/shared/utils` overtime while the rest of the transitions is being taken care of. That way, new utils / packages can rely on the new package while the old dependants are being updated
